internal/app/service/notification/service/notification.go

```diff
@@ -0,0 +1,58 @@
+package service
+
+import (
+	"proctor/internal/app/service/infra/logger"
+	"proctor/internal/app/service/infra/plugin"
+	"proctor/pkg/notification"
+	"proctor/pkg/notification/event"
+	"sync"
+)
+
+type NotificationService interface {
+	Notify(evt event.Event)
+}
+
+type notificationService struct {
+	observers           []notification.Observer
+	goPlugin            plugin.GoPlugin
+	pluginsBinary       []string
+	pluginsExportedName []string
+	once                sync.Once
+}
+
+func (s *notificationService) Notify(evt event.Event) {
+	s.initializePlugin()
+	for _, observer := range s.observers {
+		err := observer.OnNotify(evt)
```

Comment:
So the Type is there (see my other comments). But you did not filter observer by its type